app/routes.py

```diff
@@ -1,2 +1,338 @@
+    rental = Rental.query.get_or_404(rental_record_id)
+    rental.due_date = None 
+
+    db.session.commit()
+
+    return make_response({ "customer_id": updated_cust.cust_id,
+                        "video_id": updated_video.video_id,
+                        "videos_checked_out_count": updated_cust.videos_checked_out_count,
+                        "available_inventory": updated_video.available_inventory
+                    }, 
+                        200)
+
+    
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"], strict_slashes=False)
+def get_customer_rentals(customer_id):
+
+    current_rentals = []
+    current_rentals_response = []
+
+    # ❗️ concerned again about the fact that I'm not really updating the rental table at any point to indicate whether its records are current or not....
+    rental_records = Rental.query.filter_by(fk_customer_id=customer_id)
+
+    # ❗️ I know this can't be the best way to do this (might even be...the worst way), but trying to use `saved_customer.current_rentals` to get some kind of list of current rentals wasn't working in the way I expected, and I just ran out of time 
+    for each_rental in rental_records:
+        rental_vid_id = each_rental.fk_video_id
+        rental_due_date = each_rental.due_date
+        rental_id = each_rental.rental_id
+        rental_info = {"vid_id": rental_vid_id, "due_date": rental_due_date, "rental_id": rental_id}
+        current_rentals.append(rental_info)
+
+    # format the response body shown in the README
+    for each_video in current_rentals:
+        video_info = Video.query.get_or_404(each_video["vid_id"])
+        rental_info = Rental.query.get_or_404(each_video["rental_id"])
+        rented_video_info = {
+            "release_date": video_info.release_date,
+            "title": video_info.title,
+            "due_date": rental_info.due_date
+        }
+        current_rentals_response.append(rented_video_info)
+
+    return jsonify(current_rentals_response), 200
+
+
+@videos_bp.route("/<video_id>/rentals", methods=["GET"], strict_slashes=False)
+def get_video_renters(video_id):
+
+    current_renters = []
+    current_renters_response = []
+
+    # get the rental records for this video
+    rental_records = Rental.query.filter_by(fk_video_id=video_id)
+
+    # pull the info about the customers who currently have this video checked out from the video's rental records:
+    for each_rental in rental_records:
+        rental_cust_id = each_rental.fk_customer_id
+        rental_due_date = each_rental.due_date
+        rental_id = each_rental.rental_id
+        renter_info = {"customer_id": rental_cust_id, "due_date": rental_due_date, "rental_id": rental_id}
+        current_renters.append(renter_info)
+
+    # format the response body 
+    for each_renter in current_renters:
+        customer_info = Customer.query.get_or_404(each_renter["customer_id"])
+        rental_info = Rental.query.get_or_404(each_renter["rental_id"])
```

Comment:
the rental info was just retrieved above so there's no need to fetch it again here

Reply:
same goes for the similar code in the previous function